app/services/edition_unwithdrawer.rb

```diff
@@ -0,0 +1,44 @@
+class EditionUnwithdrawer < EditionService
+  def verb
+    'unwithdraw'
+  end
+
+  def past_participle
+    'unwithdrawn'
+  end
+
+  def failure_reason
+    @failure_reason ||= failure_reasons.first
+  end
+
+  def failure_reasons
+    reasons = []
+
+    reasons << "This action must be attributed to a user" if user.nil?
+    reasons << "An edition that is #{edition.current_state} cannot be #{past_participle}" unless can_transition?
+
+    reasons
+  end
+
+private
+
+  def prepare_edition
+    # The withdrawn edition have state 'published' to create a draft.
```

Comment:
I get what it's trying to say but it doesn't say it. I think it means "The withdrawn edition needs to be in the published state so as a draft can be created". ...or something like that.